hasher-matcher-actioner/hmalib/models.py

```diff
@@ -281,3 +283,56 @@ def from_time_range(
             & Key("UpdatedAt").between(start_time, end_time),
             ProjectionExpression=cls.DEFAULT_PROJ_EXP,
         ).get("Items", [])
+    @staticmethod
+    def from_dynamodb_string(string: str) -> "Label":
+        return Label(*string.split(":"))
+
+
+@dataclass
+class MatchMessage(SNSMessage):
+    """
+    Captures a match that will need to be processed. Joins the dataset and the
+    pipeline together.
+
+    Pipeline fields:
+    - `content_key`: A way for partners to refer uniquely to content on their
+      site
+    - `content_hash`: The pipeline generated hash for the content_key
+
+    Dataset fields:
+    - `banked_content_id`: Inside the bank, what's a unique way to refer to what
+      was matched against?
+    - `bank_id`: [optional][Defaults to 'threatexchange_all_collabs'] Which bank
+      did we fetch this banked_content from?
+    - `bank_source`: [optional][Defaults to 'api/threatexchange'] This is
+      forward looking, but potentially, we could have this be 'local', or
+      'api/some-other-api'
+    """
+    content_key: str
+    content_hash: str
+    banked_indicator_id: str
+
+    # source information, for now, it's okay to be hardcoded
+    # to threatexchange
```

Comment:
👍

Reply:
@BarrettOlson Just to be clear, these models pre-date your work on signal_source. I'd be happy to revisit the name for these variables.